### PR TITLE
[DEV-881] transaction persistence

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: src/integration/transaction/application/services/transaction.service.ts [hunk 1]
```diff
@@ -1,66 +1,83 @@
...
-  signed(id: string, hex: string) {
-    const tx = this.transactions.get(id);
+  async signed(id: string, hex: string) {
+    const tx = await this.repository.findOne(id);
+    const txPromise = this.transactions.get(tx.id);
     if (!tx) throw new NotFoundException('Transaction not found');
     this.transactions.delete(tx.id);
-    tx.signed(hex);
+    txPromise?.signed(hex);
+    this.repository.save(tx.signed(hex));
```
Comment (davidleomay) [inline src/integration/transaction/application/services/transaction.service.ts:48]: Missing await

File: src/integration/transaction/application/services/transaction.service.ts [hunk 2]
```diff
@@ -1,66 +1,86 @@
...
+  async invalidated(id: string, reason?: string) {
+    const tx = await this.repository.findOne(id);
+    const txPromise = this.transactions.get(tx.id);
     if (!tx) throw new NotFoundException('Transaction not found');
-    console.warn(`Invalidated ${id} with reason: ${reason}`);
     this.transactions.delete(tx.id);
-    tx.invalidated();
+    txPromise?.invalidated();
+    this.repository.save(tx.invalidated(reason));
   }
```
Comment (davidleomay) [inline src/integration/transaction/application/services/transaction.service.ts:40]: One could split the logic into map and DB related things (for all update methods), I don't know if it makes it more readable ... up to you 😉 
```ts
  async invalidated(id: string, reason?: string) {
    const tx = await this.repository.findOne(id);
    if (!tx) throw new NotFoundException('Transaction not found');
    this.repository.save(tx.invalidated(reason));

    const txPromise = this.transactions.get(tx.id);
    this.transactions.delete(tx.id);
    txPromise?.invalidated();
  }
```

Comment (Krysh90) [reply src/integration/transaction/application/services/transaction.service.ts:40]: yeah it does. I will never write code friday evening while talking to a lot of people about non related topics

File: src/integration/transaction/application/services/transaction.service.ts [hunk 3]
```diff
@@ -1,66 +1,83 @@
...
+    return this.repository.getVerified().then(this.toDto);
   }
 
-  verified(id: string, signature: string) {
-    const tx = this.transactions.get(id);
+  async verified(id: string, signature: string) {
+    const tx = await this.repository.findOne(id);
     if (!tx) throw new NotFoundException('Transaction not found');
-    tx.verifierSignature = signature;
+    this.repository.save(tx.verified(signature));
```
[the remark this answers is not in the document]
Comment (davidleomay) [reply src/integration/transaction/application/services/transaction.service.ts]: Yes, the method will just return too early.

File: src/integration/transaction/application/services/transaction.service.ts (hunk 2 above)
Comment (davidleomay) [reply src/integration/transaction/application/services/transaction.service.ts]: Haha 🤣

**Review by davidleomay — APPROVED** [review]

**Pull request merged**